test/framework/src/main/java/org/opensearch/index/shard/IndexShardTestCase.java

```diff
@@ -1301,7 +1301,7 @@ public void onReplicationFailure(
                 }
             );
             ids.add(target);
-            countDownLatch.await(1, TimeUnit.SECONDS);
+            countDownLatch.await();
```

Comment:
I think we shouldn't wait unconditionally (potentially hanging the build), having reasonable timeout is needed - let the test fail but build moves on.

Reply:
Changed to 30s with an assert after the timeout that replication completed.